### PR TITLE
[DCOS-55016]: node action dropdown, drain active nodes

### PR DESCRIPTION
**Pull request by brandonc**
Closes DCOS-55016

## Testing

1. add this key to the "uiConfiguration" key in Config.dev.ts
```
features: {
  maintenance: true
},
```
2. enable useUIConfigFixtures
3. Visit the nodes page and find the new actions dropdown on the far right (for active nodes)
4. Ensure actions don't appear when feature is disabled
5. Try draining a node (this deactivates the node as well)
6. In order to reactivate the node, issue the following curl command, but replace the Cookie header VALUE and NODE_ID tokens with your own values
```
curl -v -XPOST 'http://localhost:4200/mesos/api/v1' -H 'Content-Type: application/json' -H 'Accept: application/json' --data-binary '{"type":"REACTIVATE_AGENT", "reactivate_agent": {"agent_id": {"value": "NODE_ID"}}}' -H 'Cookie: dcos-acs-auth-cookie=VALUE; dcos-acs-info-cookie=VALUE' --compressed
```

## Notes

I'm told the drained node will eventually be garbage collected by mesos but I'm not sure how long it takes.

## Screenshots

Screen Shot 2019-07-24 at 2 35 36 PM

Screen Shot 2019-07-24 at 2 37 08 PM

### CODE REVIEW

File: tests/pages/nodes/NodesPage-cy.js [hunk 1]
```diff
@@ -1,5 +1,26 @@
 describe("Nodes Page", function() {
   context("Filters nodes table", function() {
+    function openDropdown(ipAddress) {
```
Comment (DanielMSchmidt) [inline tests/pages/nodes/NodesPage-cy.js]: Good call to move this into a helper, makes the test far more readable 👍

**Comment by brandonc** [comment]
@DanielMSchmidt thank you for removing that! I didn't realize I committed it

File: plugins/nodes/src/js/components/modals/DrainNodeModal.tsx [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        <button
+          className="button button-primary-link flush-left"
+          onClick={onClose}
+        >
+          <Trans>Cancel</Trans>
+        </button>
+        <button
+          className="button button-primary"
+          onClick={handleDrain.bind(null, props.node)}
+          disabled={false}
```
Comment (mperrotti) [inline plugins/nodes/src/js/components/modals/DrainNodeModal.tsx]: We usually disable buttons when the action is "in-flight". Is that possible here?

File: plugins/nodes/src/js/columns/NodesTableStatusColumn.tsx [hunk 3]
```diff
@@ -8,7 +8,7 @@ import Node from "#SRC/js/structs/Node";
 
 const StatusCell = React.memo(({ status }: { status: Status }) => (
   <Cell>
-    <span>
+    <span className="flex flex-align-items-center">
```
Comment (mperrotti) [inline plugins/nodes/src/js/columns/NodesTableStatusColumn.tsx]: Thanks for being a good Boy Scout here ❤️

File: package-lock.json [hunk 4]
```diff
@@ -4304,7 +4304,7 @@
     },
     "bl": {
       "version": "1.2.2",
-      "resolved": "http://registry.npmjs.org/bl/-/bl-1.2.2.tgz",
+      "resolved": "https://registry.npmjs.org/bl/-/bl-1.2.2.tgz",
```
Comment (GeorgiSTodorov) [inline package-lock.json]: Why is `package-lock` getting changed, we didn't change `package.json`?

**Comment by GeorgiSTodorov** [comment]
- Should the tooltip be on the icon, not on the node name?
Screenshot from 2019-07-29 10-46-38

File: plugins/nodes/src/js/components/DrainNodeForm.tsx [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+  maxGracePeriod: number | null;
+  decommission: boolean;
+}
+
+function DrainNodeForm(props: Props) {
+  const { formData, onChange } = props;
+
+  const helpText = (
+    <Trans>
+      The maximum amount of time allowed for tasks to gracefully terminate.
```
Comment (GeorgiSTodorov) [inline plugins/nodes/src/js/components/DrainNodeForm.tsx]: This text is not in the message catalog.

File: plugins/nodes/src/js/components/DrainNodeForm.tsx [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+  };
+
+  return (
+    <form>
+      <FormGroup>
+        <Trans render="p">Optional settings for draining node</Trans>
+        <FieldLabel>
+          <FormGroupHeading>
+            <FormGroupHeadingContent>
+              <Trans render="span">Max Grace Period</Trans>{" "}
```
Comment (GeorgiSTodorov) [inline plugins/nodes/src/js/components/DrainNodeForm.tsx]: This is also not in the message catalog.

File: plugins/nodes/src/js/components/DrainNodeForm.tsx [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+          placeholder="Enter in seconds"
+          value={formData.maxGracePeriod || ""}
+          onChange={handleChangeMaxGracePeriod}
+        />
+      </FormGroup>
+      <FormGroup>
+        <FieldLabel>
+          <FormGroupHeading>
+            <FormGroupHeadingContent>
+              <Trans render="span">Decommission</Trans>
```
Comment (GeorgiSTodorov) [inline plugins/nodes/src/js/components/DrainNodeForm.tsx]: Not in the message catalog.

File: plugins/nodes/src/js/components/DrainNodeForm.tsx [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+        </FieldLabel>
+        <FieldLabel>
+          <FieldInput
+            checked={formData.decommission}
+            name="decommission"
+            type="radio"
+            value="1"
+            onChange={handleChangeDecommission}
+          />
+          <Trans>Drain and decommission</Trans>
```
Comment (GeorgiSTodorov) [inline plugins/nodes/src/js/components/DrainNodeForm.tsx]: Not in the message catalog.

File: plugins/nodes/src/js/components/NodesTable.tsx [hunk 9]
```diff
@@ -359,7 +367,21 @@ export default class NodesTable extends React.Component<
         resizable={true}
         onResize={this.handleResize("gpu")}
         width={hasCustomWidth("gpu") ? customWidthFor("gpu") : undefined}
-      />
+      />,
+      hasMaintenance ? (
+        <Column
+          key="actions"
+          header={
```
Comment (GeorgiSTodorov) [inline plugins/nodes/src/js/components/NodesTable.tsx]: We can just do `header=""` here.

File: plugins/nodes/src/js/components/modals/DrainNodeModal.tsx [hunk 10]
```diff
@@ -0,0 +1,149 @@
+import React, { useState } from "react";
+
+// @ts-ignore
+import { request } from "@dcos/mesos-client";
+import { Trans } from "@lingui/macro";
+import { Modal } from "reactjs-components";
```
Comment (GeorgiSTodorov) [inline plugins/nodes/src/js/components/modals/DrainNodeModal.tsx]: Have we considered using the ui-kit modal here?

File: plugins/nodes/src/js/components/modals/DrainNodeModal.tsx [hunk 11]
```diff
@@ -0,0 +1,149 @@
...
+
+import Node from "#SRC/js/structs/Node";
+import ModalHeading from "#SRC/js/components/modals/ModalHeading";
+
+import DrainNodeForm, { DrainOptions } from "../DrainNodeForm";
+import { InfoBoxInline } from "@dcos/ui-kit";
```
Comment (GeorgiSTodorov) [inline plugins/nodes/src/js/components/modals/DrainNodeModal.tsx]: Mixing local and non-local imports.

File: plugins/nodes/src/js/components/modals/DrainNodeModal.tsx [hunk 12]
```diff
@@ -0,0 +1,149 @@
...
+      }).subscribe({
+        next: () => {
+          setNetworkError(null);
+          setInProgress(false);
+          onClose();
+        },
+        error: ({ code, message }: { code: number; message: string }) => {
+          setNetworkError(
+            code === 0 ? (
+              <Trans>Network is offline</Trans>
```
Comment (GeorgiSTodorov) [inline plugins/nodes/src/js/components/modals/DrainNodeModal.tsx]: Again, not in the message catalog.

File: plugins/nodes/src/js/components/modals/DrainNodeModal.tsx (hunk 10 above)
Comment (brandonc) [reply plugins/nodes/src/js/components/modals/DrainNodeModal.tsx]: I did. We don't use the ui-kit modal in dcos-ui and I figured we should migrate all at once.

**Pull request merged**